File: leetcode/editor/en/[1647]Minimum Deletions to Make Character Frequencies Unique.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {number}
 */
var minDeletions = function (s) {
    let freq = {};
    for (const c of s) {
        freq[c] = (freq[c] || 0) + 1;
    }
    let set = new Set();
    let res = 0;
    for (let num of Object.values(freq)) {
        let curr = num;
        while (curr > 0) {
            if (!set.has(curr)) {
                set.add(curr);
                break;
            }
            curr--;
            res++;
        }
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[1647]Minimum Deletions to Make Character Frequencies Unique.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int minDeletions(String s) {
        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }
        Set<Integer> set = new HashSet<>();
        int res = 0;
        for (int num : freq) {
            int curr = num;
            while (curr > 0) {
                if (!set.contains(curr)) {
                    set.add(curr);
                    break;
                }
                curr--;
                res++;
            }
        }
        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
